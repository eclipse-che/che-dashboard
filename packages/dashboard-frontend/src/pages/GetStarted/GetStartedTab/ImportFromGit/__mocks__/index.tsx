/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/GetStarted/GetStartedTab/ImportFromGit';

export class ImportFromGit extends React.PureComponent<Props> {
  public render() {
    const { hasSshKeys } = this.props;
    return (
      <div>
        <span data-testid="has-ssh-keys">{hasSshKeys ? 'true' : 'false'}</span>
      </div>
    );
  }
}
