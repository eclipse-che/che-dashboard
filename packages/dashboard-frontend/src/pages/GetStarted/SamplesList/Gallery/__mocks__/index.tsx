/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/GetStarted/SamplesList/Gallery';

export default class SamplesListGallery extends React.PureComponent<Props> {
  render() {
    const { onCardClick, metadataFiltered = [] } = this.props;
    return (
      <div data-testid="samples-list-gallery">
        <div>Samples List Gallery</div>
        {metadataFiltered.map(metadata => (
          <div key={metadata.links.v2} data-testid="sample-card">
            {metadata.displayName}

            <button
              onClick={() => {
                onCardClick(metadata);
              }}
            >
              Select Sample
            </button>
          </div>
        ))}
      </div>
    );
  }
}
