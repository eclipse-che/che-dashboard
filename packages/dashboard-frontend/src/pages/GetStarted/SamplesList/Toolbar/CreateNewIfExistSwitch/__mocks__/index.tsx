/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '..';

export class CreateNewIfExistSwitch extends React.PureComponent<Props> {
  private isChecked = false;

  render(): React.ReactNode {
    return (
      <div data-testid="create-new-if-exist">
        {this.isChecked ? 'Create New If Existing On' : 'Create New If Existing Off'}
        <input
          data-testid="create-new-if-exist-switch"
          type="checkbox"
          onClick={() => {
            this.isChecked = !this.isChecked;
            this.props.onChange(this.isChecked);
          }}
          name="Toggle Create New If Existing"
        />
      </div>
    );
  }
}
