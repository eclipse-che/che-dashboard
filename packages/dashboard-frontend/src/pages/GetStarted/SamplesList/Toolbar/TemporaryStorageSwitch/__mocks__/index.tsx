/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '..';

export default class TemporaryStorageSwitch extends React.PureComponent<Props> {
  render(): React.ReactNode {
    const { isTemporary } = this.props;
    return (
      <div>
        {isTemporary ? 'Temporary Storage On' : 'Temporary Storage Off'}
        <input
          type="checkbox"
          onClick={() => this.props.onChange(!isTemporary)}
          name="Toggle Temporary Storage"
        />
      </div>
    );
  }
}
