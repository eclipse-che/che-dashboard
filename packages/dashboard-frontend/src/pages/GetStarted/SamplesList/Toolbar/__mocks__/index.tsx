/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/GetStarted/SamplesList/Toolbar';

export default class SamplesListToolbar extends React.PureComponent<Props> {
  render(): React.ReactNode {
    const { isTemporary, onTemporaryStorageChange } = this.props;
    return (
      <div>
        <span>Samples List Toolbar</span>
        <span data-testid="isTemporary">{isTemporary ? 'true' : 'false'}</span>
        <button onClick={() => onTemporaryStorageChange(!isTemporary)}>Toggle isTemporary</button>
      </div>
    );
  }
}
