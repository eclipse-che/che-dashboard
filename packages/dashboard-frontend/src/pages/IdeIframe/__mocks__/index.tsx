/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';
import { Props } from '..';

export const iframeWorkspaceId = 'iframe-workspace-id';

export class IdeIframe extends React.Component<Props> {
  render(): React.ReactElement {
    return (
      <div data-testid="ide-iframe">
        <div data-testid="ideUrl">{this.props.ideUrl}</div>
        <div data-testid="isDevWorkspace">{this.props.isDevWorkspace}</div>
        <button onClick={() => this.props.onOpenWorkspacesList()}>Open Workspaces List</button>
        <button onClick={() => this.props.onWorkspaceRestartFromIframe(iframeWorkspaceId)}>
          Restart Workspace
        </button>
      </div>
    );
  }
}
