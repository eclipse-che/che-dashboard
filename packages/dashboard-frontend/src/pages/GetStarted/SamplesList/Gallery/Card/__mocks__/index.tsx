/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/GetStarted/SamplesList/Gallery/Card';

export class SampleCard extends React.PureComponent<Props> {
  render(): React.ReactElement {
    const { metadata, onClick } = this.props;

    return (
      <div data-testid="sample-card">
        <div data-testid="sample-name">{metadata.displayName}</div>
        <button onClick={() => onClick()}>Select Sample</button>
      </div>
    );
  }
}
