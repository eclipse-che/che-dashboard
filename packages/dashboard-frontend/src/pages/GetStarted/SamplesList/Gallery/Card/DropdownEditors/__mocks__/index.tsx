/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/GetStarted/SamplesList/Gallery/Card/DropdownEditors';

export class DropdownEditors extends React.PureComponent<Props> {
  render(): React.ReactNode {
    const { editors, onEditorSelect } = this.props;
    return (
      <div data-testid="card-actions-dropdown">
        <div>spanDropdownEditors</div>
        {editors.map(editor => (
          <button
            data-testid="card-action"
            key={editor.id}
            onClick={() => onEditorSelect(editor.id)}
          >
            {editor.name}
          </button>
        ))}
        <button onClick={() => onEditorSelect(editors[0].id)}>Select Editor</button>
      </div>
    );
  }
}
