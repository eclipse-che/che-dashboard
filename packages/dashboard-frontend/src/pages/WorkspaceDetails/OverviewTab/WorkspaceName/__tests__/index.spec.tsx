/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { ApplicationId } from '@eclipse-che/common';
import React from 'react';
import { Provider } from 'react-redux';
import { Store } from 'redux';

import WorkspaceNameLink from '@/pages/WorkspaceDetails/OverviewTab/WorkspaceName';
import getComponentRenderer from '@/services/__mocks__/getComponentRenderer';
import devfileApi from '@/services/devfileApi';
import { constructWorkspace, Workspace } from '@/services/workspace-adapter';
import { DevWorkspaceBuilder } from '@/store/__mocks__/devWorkspaceBuilder';
import { FakeStoreBuilder } from '@/store/__mocks__/storeBuilder';

const { createSnapshot } = getComponentRenderer(getComponent);

describe('WorkspaceNameLink', () => {
  let storeBuilder: FakeStoreBuilder;
  let devWorkspace: devfileApi.DevWorkspace;
  let workspace: Workspace;

  beforeEach(() => {
    devWorkspace = new DevWorkspaceBuilder().withName('my-project').build();
    workspace = constructWorkspace(devWorkspace);
    storeBuilder = new FakeStoreBuilder().withDevWorkspaces({ workspaces: [devWorkspace] });
  });

  test('screenshot when cluster console is available', () => {
    const store = storeBuilder
      .withClusterInfo({
        applications: [
          {
            id: ApplicationId.CLUSTER_CONSOLE,
            title: 'Cluster Console',
            url: 'https://console-openshift-console.apps-crc.testing',
            icon: 'icon',
          },
        ],
      })
      .build();
    const snapshot = createSnapshot(store, workspace);
    expect(snapshot.toJSON()).toMatchSnapshot();
  });

  test('screenshot when cluster console is NOT available', () => {
    const store = storeBuilder.build();
    const snapshot = createSnapshot(store, workspace);
    expect(snapshot.toJSON()).toMatchSnapshot();
  });
});

function getComponent(store: Store, workspace: Workspace) {
  return (
    <Provider store={store}>
      <WorkspaceNameLink workspace={workspace} />
    </Provider>
  );
}
