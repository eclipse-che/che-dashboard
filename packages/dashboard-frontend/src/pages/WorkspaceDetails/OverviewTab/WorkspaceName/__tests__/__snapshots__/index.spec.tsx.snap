// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkspaceNameLink screenshot when cluster console is NOT available 1`] = `
<div
  className="pf-c-form__group"
>
  <div
    className="pf-c-form__group-label"
  >
    <label
      className="pf-c-form__label"
    >
      <span
        className="pf-c-form__label-text"
      >
        Workspace
      </span>
    </label>
     
  </div>
  <div
    className="pf-c-form__group-control"
  >
    <span
      className="readonly"
    >
      my-project
    </span>
  </div>
</div>
`;

exports[`WorkspaceNameLink screenshot when cluster console is available 1`] = `
<div
  className="pf-c-form__group"
>
  <div
    className="pf-c-form__group-label"
  >
    <label
      className="pf-c-form__label"
    >
      <span
        className="pf-c-form__label-text"
      >
        Workspace
      </span>
    </label>
     
  </div>
  <div
    className="pf-c-form__group-control"
  >
    <span
      className="readonly"
    >
      <a
        aria-disabled={false}
        aria-label={null}
        className="pf-c-button pf-m-link pf-m-inline"
        data-ouia-component-id="OUIA-Generated-Button-link-1"
        data-ouia-component-type="PF4/Button"
        data-ouia-safe={true}
        disabled={null}
        href="https://console-openshift-console.apps-crc.testing/k8s/ns//workspace.devfile.io~v1alpha2~DevWorkspace/my-project"
        role={null}
        target="_blank"
        type={null}
      >
        my-project
        <span
          className="pf-c-button__icon pf-m-end"
        >
          <svg
            aria-hidden={true}
            aria-labelledby={null}
            fill="currentColor"
            height="1em"
            role="img"
            style={
              {
                "verticalAlign": "-0.125em",
              }
            }
            viewBox="0 0 448 512"
            width="1em"
          >
            <path
              d="M448 80v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48zm-88 16H248.029c-21.313 0-32.08 25.861-16.971 40.971l31.984 31.987L67.515 364.485c-4.686 4.686-4.686 12.284 0 16.971l31.029 31.029c4.687 4.686 12.285 4.686 16.971 0l195.526-195.526 31.988 31.991C358.058 263.977 384 253.425 384 231.979V120c0-13.255-10.745-24-24-24z"
            />
          </svg>
        </span>
      </a>
    </span>
  </div>
</div>
`;
